torch/utils/data/datapipes/iter/callable.py

```diff
@@ -151,6 +151,9 @@ def __len__(self) -> int:
         )
 
     def __getstate__(self):
+        if IterDataPipe.getstate_hook is not None:
+            return IterDataPipe.getstate_hook(self)
+
```

Comment:
This is needed for all subclasses of `IterDataPipe` which implement `__getstate__`